scripts/build-tools.sh

```diff
@@ -146,12 +146,20 @@ main()
                 exit 0
         fi
 
-        # Keep 'topologies' first because it's the noisiest.
-        for util in topologies tests; do
-                if eval '$DO_BUILD_'$util; then
-                        make_tool $util
-                fi
-        done
+        if eval $DO_BUILD_topologies; then
+                make_tool topologies
+
+                TPLG2_RELEASE_PATH="$BUILD_TOOLS_DIR/topology/topology2/production/target/"
+                echo ""; echo "Target directory for topology2: $TPLG2_RELEASE_PATH"
+                mkdir -p "$TPLG2_RELEASE_PATH/sof-ipc4-tplg"
+                mv $BUILD_TOOLS_DIR/topology/topology2/production/*.tplg $TPLG2_RELEASE_PATH/sof-ipc4-tplg/
```

Comment:
Finally noticing this, sorry for the belated review.

```suggestion
                mv $BUILD_TOOLS_DIR/topology/topology2/production/*.tplg $TPLG2_RELEASE_PATH/sof-ipc4-tplg/
```

Nope.

Never move, delete or rename the files that CMake/ninja just built. They belong to CMake/ninja; `$BUILD_TOOLS_DIR` is EXCLUSIVE property, don't trespass with some outsider script unknown to CMake.  This will cause at least these problems:
- spurious rebuilds
- invoking ninja directly behaves differently from invoking this script. Debugging layers of indirections is painful enough, in no circumstances should this behave differently.
- other, unknown and unexpected CMake issues because no one ever does that ever pulls the rug under CMake/ninja like this.

These files can be at most copied but the best is really to structure the source directories to match what you want the build directories to be. In other words: move and rename the sources to match.

I hate git renames more than the average but this one case where they must be performed.

If you really, really, really can't move sources then look at CMake's "install" feature. Spoiler alert: you don't want that.

Reply:
> I hate git renames more than the average but this one case where they must be performed.

That does not mean moving **everything**. Moving only the CMakeLists.txt file and changing the `include()` should work. Or tricks like `CMAKE_CURRENT_BINARY_DIR/../better_dir/` might work too. As long as it does not mess with the build directory _behind CMake's back_ then it's worth a try.

On the other hand, renaming everything would be more consistent with your overarching and admirable quest for... consistency and simplification.